aws-android-sdk-core/src/main/java/com/amazonaws/auth/CognitoCredentialsProvider.java

```diff
@@ -440,6 +440,16 @@ public void setSessionCredentialsExpiration(Date expiration) {
         }
     }
 
+    public Date getSessionCredentialsExpiration() {
+        credentialsLock.readLock().lock();
+        try {
+            return sessionCredentialsExpiration;
+        } finally {
+            credentialsLock.readLock().unlock();
+        }
+    }
+
+    @Deprecated
     public Date getSessionCredentitalsExpiration() {
```

Comment:
This method should call the new method, to avoid having the implementation logic duplicated in two places.

Reply:
```java
public void newMethodWithGoodName() {
    // Contains the full implementation, that we will keep forever.
}

@Deprecated 
public void oldAndBadlyNamedMethod() {
    // It just forwards to the good method.
    methodWithCorrectName();
}
```